### PR TITLE
Add All Output Data to DLL Interface

### PR DESCRIPTION
**Pull request by dzalkind**
This PR is not ready to merge

**Feature or improvement description**
Send all OpenFAST outputs to DLL interface 

**Related issue, if one exists**

**Impacted areas of the software**
`BladedInteface_Ex`
`ServoDyn`
`FAST_Subs`
`FAST_Solver`

**Additional supporting information**

**Test results, if applicable**

### CODE REVIEW

File: modules/openfast-library/src/FAST_Solver.f90 [hunk 1]
```diff
@@ -885,10 +885,11 @@ END SUBROUTINE AD14_InputSolve_NoIfW
 
 !----------------------------------------------------------------------------------------------------------------------------------
 !> This routine sets the inputs required for ServoDyn
-SUBROUTINE SrvD_InputSolve( p_FAST, m_FAST, u_SrvD, y_ED, y_IfW, y_OpFM, y_BD, y_SD, MeshMapData, ErrStat, ErrMsg )
+SUBROUTINE SrvD_InputSolve( p_FAST, y_FAST, m_FAST, u_SrvD, y_ED, y_IfW, y_OpFM, y_BD, y_SD, MeshMapData, ErrStat, ErrMsg )
 !..................................................................................................................................
 
    TYPE(FAST_ParameterType),         INTENT(IN)     :: p_FAST       !< Glue-code simulation parameters
+   TYPE(FAST_OutputFileType),        INTENT(INOUT)  :: y_FAST              !< Output variables for the glue code
```
Comment (dzalkind) [inline modules/openfast-library/src/FAST_Solver.f90:892]: TODO: Change this to AllOuts only

File: modules/openfast-library/src/FAST_Subs.f90 [hunk 2]
```diff
@@ -2112,6 +2136,15 @@ SUBROUTINE FAST_InitOutput( p_FAST, y_FAST, Init, ErrStat, ErrMsg )
    
    NumOuts   = SUM( y_FAST%numOuts )
 
+   ! If already allocated from previous call, deallocate?
+   IF (ALLOCATED(y_FAST%ChannelNames)) THEN
+      DEALLOCATE(y_FAST%ChannelNames)
+   ENDIF 
+
+   IF (ALLOCATED(y_FAST%ChannelUnits)) THEN
```
Comment (dzalkind) [inline modules/openfast-library/src/FAST_Subs.f90:2144]: We could pre-emptively deallocate before returning the first time.  Might be less hacky.

File: modules/servodyn/src/BladedInterface_EX.f90 [hunk 3]
```diff
@@ -206,9 +211,20 @@ subroutine InitPtfmMotionSensors()     ! Channels 1001:1018
    end subroutine InitPtfmMotionSensors
 
 
-   subroutine InitNonLidarSensors()    ! Channels 1019:2000
+   subroutine InitNonLidarSensors()    ! Channels 1019:2000, 4001:5000
```
Comment (dzalkind) [inline modules/servodyn/src/BladedInterface_EX.f90:214]: TODO: make new subroutine

File: modules/servodyn/src/BladedInterface_EX.f90 [hunk 4]
```diff
@@ -474,6 +493,14 @@ subroutine SetEXavrSWAP_Sensors()
       ! Set other sensors here (non-lidar measurements)
       !       Add summary file descriptions about channels to InitNonLidarSensors as channels are added.
 
+      ! Add OpenFAST outputs
+      ChanInd = OutData_StartIdx
+
+      ! Limit number of channels to OutData_MaxChan, u%Channel names has Time, AllOutData does not
+      N_Channels = MIN(Size(u%ChannelNames)-1,OutData_MaxChan)   
+      
+      dll_data%avrSWAP(ChanInd+1:ChanInd+N_Channels) = u%AllOutData(:,u%n_Out)   ! u%n_Out is output time index
```
Comment (dzalkind) [inline modules/servodyn/src/BladedInterface_EX.f90:502]: `1:N_Channels` on RHS

File: modules/openfast-library/src/FAST_Solver.f90 [hunk 5]
```diff
@@ -989,6 +990,11 @@ SUBROUTINE SrvD_InputSolve( p_FAST, m_FAST, u_SrvD, y_ED, y_IfW, y_OpFM, y_BD, y
    u_SrvD%LSShftFys = y_ED%LSShftFys
    u_SrvD%LSShftFzs = y_ED%LSShftFzs
 
+
+   ! Set inputs here!!
+   u_SrvD%AllOutData = y_FAST%AllOutData
```
Comment (dzalkind) [inline modules/openfast-library/src/FAST_Solver.f90:995]: Pass only the last index of `y_FAST%AllOutData` and make 1D.  Can drop `n_out` then.

**Comment by andrew-platt** [comment]
A new method has been proposed using ZeroMQ to pass channels to the controller.  Closing this PR while we pursue that method instead.

**Pull request closed without merging**